pkg/synchronizer/githubsynchronizer/synchronizer.go

```diff
@@ -320,25 +320,27 @@ func (sync *ChannelSynchronizer) handleResourceDeployable(
 
 	// Update deployables if the template is updated in the git repo
 	newdpl := newDplList[dpl.Name]
-	newdpltpl := &unstructured.Unstructured{}
-	err = json.Unmarshal(newdpl.Spec.Template.Raw, newdpltpl)
-
-	if err != nil {
-		klog.Warning("Failed to unmarshall the template from updated deployable:", newdpl, err)
-		return errors.Wrap(err, fmt.Sprintf("failed to unmarshall the template from updated deployable: %v", newdpl))
-	}
-
-	if !reflect.DeepEqual(dpltpl, newdpltpl) {
-		klog.Info("Sync - Updating the template in existing deployable ", dpl.Name, " in channel ", chn.Name)
-		dpl.Spec.Template.Raw, err = json.Marshal(newdpltpl)
+	if newdpl != nil {
+		newdpltpl := &unstructured.Unstructured{}
+		err = json.Unmarshal(newdpl.Spec.Template.Raw, newdpltpl)
 
 		if err != nil {
-			klog.Warning(err.Error())
-			return errors.Wrap(err, fmt.Sprintf("failed to mashall template %v", newdpltpl))
+			klog.Warning("Failed to unmarshall the template from updated deployable:", newdpl, err)
+			return errors.Wrap(err, fmt.Sprintf("failed to unmarshall the template from updated deployable: %v", newdpl))
 		}
 
-		if err := sync.kubeClient.Update(context.TODO(), &dpl); err != nil {
-			return errors.Wrap(err, fmt.Sprintf("failed to update deployable %v", dpl.Name))
+		if !reflect.DeepEqual(dpltpl, newdpltpl) {
+			klog.Info("Sync - Updating the template in existing deployable ", dpl.Name, " in channel ", chn.Name)
+			dpl.Spec.Template.Raw, err = json.Marshal(newdpltpl)
+
+			if err != nil {
+				klog.Warning(err.Error())
+				return errors.Wrap(err, fmt.Sprintf("failed to mashall template %v", newdpltpl))
+			}
+
+			if err := sync.kubeClient.Update(context.TODO(), &dpl); err != nil {
+				return errors.Wrap(err, fmt.Sprintf("failed to update deployable %v", dpl.Name))
```

Comment:
since we wrap this error, I guess over here,

https://github.com/open-cluster-management/multicloud-operators-channel/blob/4f679478c6d8265d2c5ac78d75cfe2520a6ddbd2/pkg/synchronizer/githubsynchronizer/synchronizer.go#L168-L170

It would be nicer, if the caller print out the whole error stack,
Instead of `klog.Error(errors.Cause(err).Error())`,we can do `klog.Errorf("failed to handle single deployable, err %+v", err)`

FYI, https://godoc.org/github.com/pkg/errors#hdr-Formatted_printing_of_errors

how do you think?

Reply:
Good idea Ian. Made the change.